Improvement: Add NPC sell toggle for Not Clickable Items

lunaynx opened the pull request:
## What
Adjusted the Not Clickable Items feature so that by default, in NPC trade menus, only items that cannot be sold or are recombobulated are grayed out, but added a default off toggle to also gray out items that "should not" be sold (our whitelist for that is quite outdated and hasn't been touched in ages, but it can be improved in the future).

## Changelog Improvements
+ Improved the "Not Clickable Items" feature in NPC trade menus. - Luna
    * By default, it only grays out items that cannot be sold to NPCs, or are recombobulated.
    * You can re-enable graying out items that are not commonly sold to NPCs in the config.

lunaynx committed:
Improvement: Add NPC sell toggle for Not Clickable Items

lunaynx committed:
Further clean up logic a bit

lunaynx committed:
Add museum check

lunaynx committed:
I don't really care, stupid rule

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/inventory/HideNotClickableItems.kt:
```diff
@@ -482,25 +485,31 @@ object HideNotClickableItems {
+            return true
+        }
 
+        if (config.protectRarelySoldItems) {
```
could use early return

hannibal002 reviewed (changes requested):
i like the changes!

for the future, we could compare npc price with bz/ah price and only show if the diff per/percentage wise is "worth enough" to sell via npc instead

lunaynx committed:
Use early return

lunaynx committed:
Use "Click to sell" line to detect sellability

More futureproof, and currently some items like runes don't resolve the
NPC price properly.

lunaynx committed:
Fix imports

lunaynx committed:
Fix pattern
